kroxylicious-schema-validation/src/main/java/io/kroxylicious/proxy/filter/schema/ProduceValidationFilter.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.proxy.filter.schema;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.kafka.common.message.ProduceRequestData;
+import org.apache.kafka.common.message.ProduceResponseData;
+import org.apache.kafka.common.message.RequestHeaderData;
+import org.apache.kafka.common.message.ResponseHeaderData;
+import org.apache.kafka.common.protocol.Errors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.kroxylicious.proxy.filter.KrpcFilterContext;
+import io.kroxylicious.proxy.filter.ProduceRequestFilter;
+import io.kroxylicious.proxy.filter.ProduceResponseFilter;
+import io.kroxylicious.proxy.filter.schema.validation.ProduceRequestValidator;
+import io.kroxylicious.proxy.filter.schema.validation.RecordValidationFailure;
+import io.kroxylicious.proxy.filter.schema.validation.TopicValidationResult;
+import io.kroxylicious.proxy.filter.schema.validation.ValidationResult;
+
+import static java.util.stream.Collectors.joining;
+
+public class ProduceValidationFilter
+        implements ProduceRequestFilter, ProduceResponseFilter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ProduceValidationFilter.class);
+    private final ProduceRequestValidator validator;
+    private final Map<Integer, ValidationResult> correlatedResults = new HashMap<>();
+
+    public ProduceValidationFilter(ProduceRequestValidator validator) {
+        this.validator = validator;
+    }
+
+    @Override
+    public void onProduceRequest(RequestHeaderData header, ProduceRequestData request, KrpcFilterContext context) {
+        ValidationResult result = validator.validateRequest(request);
+        if (result.allTopicPartitionsInvalid()) {
+            LOGGER.warn("all topic-partitions for request contained invalid data: {}", result);
```

Comment:
IMO this isn't a `warn`. You need to distinguish what warn is for. This situation happens entirely _normally_, as a result of some application sending invalid data, and the error reporting mechanism for that is the return of a `ProduceResponse` to the application. So when this happens it does not indicate a problem in this filter at all, it's just the filter doing its job. So it shouldn't be a warn. It's `info` at best, or more likely `debug`.